### PR TITLE
swarm free feeds when unique feed generation is disabled

### PR DESCRIPTION
**Pull request by mafintosh**
This small change, makes any dazaar feed that is free and has uniqueFeed generation disabled will swarm like normal hypercores as well, which makes it scale very nicely :)

### CODE REVIEW

**Comment by robertkowalski** [comment]
on the serverside after restarting the client:

```
$ node import.js --pair=BTCUSD --type=trades
~/bitfinex/dazaar-bfx-market-data(node:19009) UnhandledPromiseRejectionWarning: Error: Block not downloaded
    at Feed.get (/Users/robert/bitfinex/dazaar-bfx-market-data/node_modules/hypercore/index.js:1248:66)
    at Promise (/Users/robert/bitfinex/dazaar-bfx-market-data/node_modules/hyperbee/index.js:324:32)
    at new Promise (<anonymous>)
    at HyperBee.getBlock (/Users/robert/bitfinex/dazaar-bfx-market-data/node_modules/hyperbee/index.js:322:12)
    at Batch.getBlock (/Users/robert/bitfinex/dazaar-bfx-market-data/node_modules/hyperbee/index.js:467:25)
    at HyperBee.getRoot (/Users/robert/bitfinex/dazaar-bfx-market-data/node_modules/hyperbee/index.js:314:25)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

File: swarm.js [hunk 1]
```diff
@@ -22,7 +22,10 @@ function swarm (m, onjoin, opts) {
   const announce = market.isSeller(m)
   const lookup = !announce
 
-  m.ready(() => swarm.join(m.discoveryKey, { announce, lookup }, onjoin))
+  m.ready(() => {
+    if (announce && !m.uniqueFeed) swarm.join(m.feed.discoveryKey, { announce, lookup })
```
Comment (robertkowalski) [inline swarm.js:26]: is an early return required?

Comment (robertkowalski) [reply swarm.js:26]: hmmm... on a uniquefeed `m.uniqueFeed === undefined` for me.

here is my setup:

```js
const dopts = { masterKey: keys.masterKey }
const market = dazaar(path.join(__dirname, 'dbs', 'dazaar', name), dopts)

let feed
market.ready(() => {
  feed = hypercore(
    path.join(__dirname, 'dbs', 'core', name),
    keyPair.publicKey,
    { stats, secretKey: keyPair.secretKey }
  )

  feed.ready(next)
})


function next () {
  seller = market.sell(feed, {
    validate (remoteKey, done) {
      payment.validate(remoteKey, function (err, info) {
        done(err, info)
      })
    },
    uniqueFeed: false
  })
}
```

Comment (mafintosh) [reply swarm.js:26]: Thanks, will investigate

Comment (mafintosh) [reply swarm.js]: I think this is solved now with a reinstall (was a hyperbee bug)

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**